### PR TITLE
Industry name in JobSkills Algolia Serializer

### PR DESCRIPTION
**Pull request by bilaltahir21**
# Discussion
This PR adds industry names in the industry_names field in `JobSerializer` which contains the industry names in which the job belongs. The Jira ticket has decided structure of serializer. 

JIRA: [ENT-6411](https://2u-internal.atlassian.net/browse/ENT-6411)

## Algolia Index
Some of the screenshots of Algolia index with data from local database. 

Job which is not linked to any industry. 
Screenshot 2022-11-23 at 2 18 41 PM

Jobs which are linked to some industries. 
Screenshot 2022-11-23 at 2 04 26 PM

Screenshot 2022-11-23 at 2 02 02 PM

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/algolia/serializers.py [hunk 1]
```diff
@@ -79,6 +81,16 @@ def get_job_postings(self, obj):
...
 
+    def get_industry_names(self, obj):
+        """
+        Get a list of industry names associated with the job.
+
+        Arguments:
+            obj (Job): Job instance whose industries need to be fetched.
+        """
+        qs = IndustryJobSkill.objects.filter(job=obj).select_related('industry')
+        return [industry.industry.name for industry in qs]
```
Comment (sameenfatima78) [inline taxonomy/algolia/serializers.py:92]: nit: can we rename the loop variable from `industry` to `job_industry`?

**Review by sameenfatima78 — COMMENTED** [review]
Looks good but we need to add test case for this new change

File: taxonomy/algolia/serializers.py (hunk 1 above)
Comment (bilaltahir21) [reply taxonomy/algolia/serializers.py]: Renamed it to `industry_job_skill` because it is being queried from the model `IndustryJobSkill `.

Comment (bilaltahir21) [reply taxonomy/algolia/serializers.py]: Added this change in the test.

**Review by sameenfatima78 — APPROVED** [review]

**Pull request merged**